src/svsm.rs

```diff
@@ -449,31 +449,43 @@ pub extern "C" fn svsm_main() {
 
     start_secondary_cpus(&cpus);
 
-    let fw_meta = map_and_parse_fw_meta()
-        .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e));
+    let fw_metadata: Option<SevFWMetaData>;
+    if config.should_launch_fw() {
+        fw_metadata = Some(
+            map_and_parse_fw_meta()
+            .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e)));
+    } else {
+        fw_metadata = None;
+    }
```

Comment:
The `fw_metadata` can be constructed directly:

```suggestion
    let fw_metadata = if config.should_launch_fw() {
        Some(
            map_and_parse_fw_meta()
                .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e)),
        )
    } else {
        None
    };
```

Or even better:

```suggestion
    let fw_metadata = config.should_launch_fw().then(|| {
        map_and_parse_fw_meta()
            .unwrap_or_else(|e| panic!("Failed to parse FW SEV meta-data: {:#?}", e))
    });
```